packages/cli/src/tools/config/index.js

```diff
@@ -97,90 +97,94 @@ function loadConfig(projectRoot: string = process.cwd()): ConfigT {
 
   let depsWithWarnings = [];
 
-  const finalConfig = findDependencies(projectRoot).reduce(
-    (acc: ConfigT, dependencyName) => {
-      let root;
-      let config;
-      try {
-        root = resolveNodeModuleDir(projectRoot, dependencyName);
-        const output = readDependencyConfigFromDisk(root);
-        config = output.config;
+  const finalConfig = [
+    ...Object.keys(userConfig.dependencies),
+    ...findDependencies(projectRoot),
+  ].reduce((acc: ConfigT, dependencyName) => {
+    const localDependencyRoot =
+      userConfig.dependencies[dependencyName] &&
+      userConfig.dependencies[dependencyName].root;
```

Comment:
There's also one conceptual thing regarding `userConfig` that I really wanted to avoid. By default, user configuration is deep merged with the configuration we find out. While finding out the proper configuration, we take dependency config into consideration. 

User provided configuration can overwrite any part of the dependency configuration but that will not affect other properties. For example, you can overwrite the android package import path, but this will not affect other properties. 

Taking `root` into consideration breaks this contract.

Also, have we tested for duplicate keys?

Reply:
What if we exported a utility function instead and asked users to do:
```
module.exports = {
   dependencies: {
      'my-custom-dependency': resolveDependencyINPath('...')
   },
};
```
that would still preserve the merging approach we have settled on so far and actually, give us an opportunity to extract some pieces of this file into reusable functions?

Reply:
> User provided configuration can overwrite any part of the dependency configuration but that will not affect other properties. For example, you can overwrite the android package import path, but this will not affect other properties.
Taking root into consideration breaks this contract

Not really sure about that. `root` is kinda special, because it lets CLI resolve other values, but you can still override them:
```js
const root = process.cwd();

module.exports = {
  dependencies: {
    'react-native-slider': {
      root,
      platforms: {
        ios: {
          sourceDir: 'lala',
          libraryFolder: 'lol/lib',
        },
      },
    },
  },
};
```

resolves to:

```json5
{
  "dependencies": {
    "react-native-slider": {
      "root": "/Users/thymikee/Projects/oss/react-native-slider",
      "name": "react-native-slider",
      "platforms": {
        "ios": {
          "sourceDir": "lala",
          // regular params
          "libraryFolder": "lol/lib",
        },
        "android": {
          // regular params
        }
      },
      "assets": [],
      "hooks": {},
      "params": []
    }
  }
}
```

That's why I chose this solution. It's "nothing new" and user is still able to override things.

Reply:
I included above in a test so it doesn't go away